### PR TITLE
[ah_var_store] Retry with more memory VCF extract [VS-1351]

### PR DESCRIPTION
**Pull request by mcovarr**
~Currently in draft status, I still need to test this the way I tested the PGEN version.~ Tested, works, ready for review!

### CODE REVIEW

File: scripts/variantstore/wdl/GvsExtractCallset.wdl [hunk 1]
```diff
@@ -368,7 +368,22 @@ task ExtractTask {
         --filter-set-name ~{filter_set_name}'
     fi
 
-    gatk --java-options "-Xmx~{memory_gib - 3}g" \
+    # This tool may get invoked with "Retry with more memory" with a different amount of memory than specified in
+    # the input `memory_gib`, so use the memory-related environment variables rather than the `memory_gib` input.
```
Comment (gbggrant) [inline scripts/variantstore/wdl/GvsExtractCallset.wdl:372]: Isn't there a way to set the memory in the run-time block from the memory-related environment variables?

Comment (mcovarr) [reply scripts/variantstore/wdl/GvsExtractCallset.wdl:372]: Not that I know of, and the way "retry with more memory" works I don't think it would matter. Cromwell basically remembers how much memory the task was last run with, and if it fails with looks like an out of memory error, Cromwell retries the task with an overridden, "multiplied" memory value: https://support.terra.bio/hc/en-us/articles/4403215299355-Out-of-Memory-Retry

Comment (gbggrant) [reply scripts/variantstore/wdl/GvsExtractCallset.wdl:372]: Huh. But that would defeat the whole point. So it just allocates to the VM the same amount of memory (here memory_gib) each time it retries??
Line 1.2 `Some commands/tools might be able to pick up and use the additional memory provided automatically without customization.` seems to imply that the VM will come up with more memory and tools like python would then recognize that they're on a VM with more memory than before?.

Comment (gbggrant) [reply scripts/variantstore/wdl/GvsExtractCallset.wdl:372]: Oh, I see: https://cromwell.readthedocs.io/en/stable/cromwell_features/RetryWithMoreMemory/

Comment (mcovarr) [reply scripts/variantstore/wdl/GvsExtractCallset.wdl:372]: [Here's a run](https://app.terra.bio/#workspaces/gvs-dev/GVS%20Integration/job_history/7fe279de-ab36-448f-b0c2-1452349d0319) where I added in some fake "out of memory" code that shows the task being retried with increasing amounts of memory.

Comment (gbggrant) [reply scripts/variantstore/wdl/GvsExtractCallset.wdl:375]: Ah, I see that you had it OOM before hitting the actual Java command, and from try 4: `MEM_SIZE: 69.12, MEM_UNIT: GB, memory_mb: 66120, memory_gib: 40`.

So, what I'm still confused about is how would cromwell up the memory needed for the VM for each retry? At this point the memory that Java will ask for is more than the VM has. The VM's memory is *always* set to memory_gib which is 40. How is that even supposed to work? Is it possible that if the runtime memory was hard-coded to 40, then cromwell might be smart enough to up that as it's upping MEM_SIZE?

Basically, I don't understand the use of retrying with more memory if there's no way that that additional memory isn't requested on the VM we are spinning up.

Comment (mcovarr) [reply scripts/variantstore/wdl/GvsExtractCallset.wdl:375]: Basically Cromwell doesn't honor the original `memory` specification on retry-with-more-memory retries, it uses the scaled values instead.

**Comment by mcovarr** [comment]
Tested and working, taking out of Draft status.

**Review by rsasch — APPROVED** [review]
LGTM.  Also wondering if we want to add in documentation on the quota increases that we need for extract so that those can be requested well before extract is kicked off?

**Comment by mcovarr** [comment]
> LGTM. Also wondering if we want to add in documentation on the quota increases that we need for extract so that those can be requested well before extract is kicked off?

I will be more than happy to document how to have quotas increased as soon as I figure out how to actually accomplish that, which so far I have not. 😞
